apiserver/src/api.rs

```diff
@@ -1,95 +1,126 @@
+        let node_name = extract_header_string(headers, HEADER_BRUPOP_NODE_NAME)?;
+        let node_uid = extract_header_string(headers, HEADER_BRUPOP_NODE_UID)?;
+        let _k8s_auth_token = extract_header_string(headers, HEADER_BRUPOP_K8S_AUTH_TOKEN)?;
+
+        Ok(ApiserverCommonHeaders {
+            node_selector: BottlerocketNodeSelector {
+                node_name,
+                node_uid,
+            },
+            _k8s_auth_token,
+        })
+    }
+}
 
 /// HTTP endpoint that implements a shallow health check for the HTTP service.
 async fn health_check() -> impl Responder {
     HttpResponse::Ok().body("pong")
 }
 
-#[derive(Debug, Clone, Serialize, Deserialize)]
-/// Describes a node for which a BottlerocketNode custom resource should be constructed.
-pub struct CreateBottlerocketNodeRequest {
-    pub node_selector: BottlerocketNodeSelector,
-}
-
 /// HTTP endpoint which creates BottlerocketNode custom resources on behalf of the caller.
 async fn create_bottlerocket_node_resource<T: BottlerocketNodeClient>(
     req: HttpRequest,
     settings: web::Data<APIServerSettings<T>>,
-    create_request: web::Json<CreateBottlerocketNodeRequest>,
+    http_req: HttpRequest,
     root_span: RootSpan,
 ) -> Result<impl Responder> {
-    root_span.record(
-        "node_name",
-        &create_request.node_selector.node_name.as_str(),
-    );
-    inner_create_bottlerocket_node_resource(req, settings, create_request).await
+    let headers = ApiserverCommonHeaders::try_from(http_req.headers())?;
+    root_span.record("node_name", &headers.node_selector.node_name.as_str());
+    inner_create_bottlerocket_node_resource(req, settings, http_req).await
 }
 
 /// Inner implementation of `create_bottlerocket_node_resource`. Provided so that unit tests can avoid setting up
 /// tracing loggers, which make it difficult to introspect HTTP responses in test cases.
 async fn inner_create_bottlerocket_node_resource<T: BottlerocketNodeClient>(
     _req: HttpRequest,
     settings: web::Data<APIServerSettings<T>>,
-    create_request: web::Json<CreateBottlerocketNodeRequest>,
+    http_req: HttpRequest,
 ) -> Result<impl Responder> {
+    let headers = ApiserverCommonHeaders::try_from(http_req.headers())?;
     let br_node = settings
         .node_client
-        .create_node(&create_request.node_selector)
+        .create_node(&headers.node_selector)
         .await
         .context(error::BottlerocketNodeCreate)?;
 
     Ok(HttpResponse::Ok().body(format!("{}", json!(&br_node))))
 }
 
-#[derive(Debug, Clone, Serialize, Deserialize)]
-/// Describes updates to a BottlerocketNode object's `status`.
-pub struct UpdateBottlerocketNodeRequest {
-    pub node_selector: BottlerocketNodeSelector,
-    pub node_status: BottlerocketNodeStatus,
-}
-
 /// HTTP endpoint which updates the `status` of a BottlerocketNode custom resource on behalf of the caller.
 async fn update_bottlerocket_node_resource<T: BottlerocketNodeClient>(
```

Comment:
Actually when I read this file, I have a feeling that it might be better if we put this api.rs to a folder and divide each handler to a separate file, I am not sure if this is suitable for rust code structure arrange though

Reply:
It can certainly be done! Sometimes traversing more files can be a nuisance, but there's not much coupling here so I think it should be ok. I'll try it.

Reply:
I've implemented this in #124